shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/screens/TermsOfUse.kt

```diff
@@ -1,21 +1,24 @@
 package org.jetbrains.kotlinconf.screens
 
-import androidx.compose.foundation.Image
-import androidx.compose.foundation.clickable
-import androidx.compose.foundation.layout.Column
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
 import kotlinconfapp.shared.generated.resources.Res
-import kotlinconfapp.shared.generated.resources.arrow_left_24
-import org.jetbrains.compose.resources.painterResource
+import kotlinconfapp.shared.generated.resources.general_terms
+import kotlinconfapp.shared.generated.resources.visitors_terms_title
+import kotlinconfapp.shared.generated.resources.visitors_terms_version
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.stringResource
 import org.jetbrains.kotlinconf.LocalNavController
-import org.jetbrains.kotlinconf.ui.components.StyledText
+import org.jetbrains.kotlinconf.ui.components.MarkdownScreenWithTitle
 
+@OptIn(ExperimentalResourceApi::class)
 @Composable
 fun TermsOfUse() {
     val navController = LocalNavController.current
-    Column {
-        Image(painterResource(Res.drawable.arrow_left_24), "back", modifier = Modifier.clickable { navController.popBackStack() })
-        StyledText("Terms of Use")
-    }
-}
\ No newline at end of file
+    MarkdownScreenWithTitle(
+        title = stringResource(Res.string.general_terms),
+        header = stringResource(Res.string.visitors_terms_title),
+        subheader = stringResource(Res.string.visitors_terms_version),
+        loadText = { Res.readBytes("files/visitors-terms.md") },
+        onBack = navController::popBackStack
```

Comment:
The `NavController` passed down through composition locals makes me uncomfortable, as it tightly couples the individual screens to the nav setup, and also spreads pieces of navigation logic all over these screens.

What do you think about routing these navigation events back up to the `App` level so that they can be handled there, centrally?

It's what I did for screens on https://github.com/JetBrains/kotlinconf-app/pull/223/files